iothub/device/src/IotHubConnection.cs

```diff
@@ -473,29 +465,15 @@ static AmqpSettings CreateAmqpSettings()
             return amqpSettings;
         }
 
-        protected static AmqpLinkSettings SetLinkSettingsCommonProperties(AmqpLinkSettings linkSettings, TimeSpan timeSpan, ProductInfo productInfo)
+        protected static AmqpLinkSettings SetLinkSettingsCommonProperties(AmqpLinkSettings linkSettings, TimeSpan timeSpan, ProductInfo productInfo, string corrId, string linkType)
         {
             linkSettings.AddProperty(IotHubAmqpProperty.TimeoutName, timeSpan.TotalMilliseconds);
-
             linkSettings.AddProperty(IotHubAmqpProperty.ClientVersion, productInfo.ToString());
+            linkSettings.AddProperty(IotHubAmqpProperty.ChannelCorrelationId, linkType + ":" + corrId);
 
             return linkSettings;
         }
 
-        protected static AmqpLinkSettings SetLinkSettingsCommonPropertiesForMethod(AmqpLinkSettings linkSettings, string corrId)
-        {
-            linkSettings.AddProperty(IotHubAmqpProperty.ApiVersion, ClientApiVersionHelper.ApiVersionString);
-            linkSettings.AddProperty(IotHubAmqpProperty.ChannelCorrelationId, "methods:" + corrId);
-            return linkSettings;
-        }
-
-        AmqpLinkSettings SetLinkSettingsCommonPropertiesForTwin(AmqpLinkSettings linkSettings, string corrId)
-        {
-            linkSettings.AddProperty(IotHubAmqpProperty.ApiVersion, ClientApiVersionHelper.ApiVersionString);
-            linkSettings.AddProperty(IotHubAmqpProperty.ChannelCorrelationId, "twin:" + corrId);
-            return linkSettings;
```

Comment:
is it ok to ommit "twin:" or "methods:" from correlation id ?

Reply:
Why was it needed in the first place ?

Reply:
Those values are still being provided in the new common correlation id setter:
```cSharp
 protected static AmqpLinkSettings SetLinkSettingsCommonProperties(AmqpLinkSettings linkSettings, TimeSpan timeSpan, ProductInfo productInfo, string corrId, string linkType)
{
...
linkSettings.AddProperty(IotHubAmqpProperty.ChannelCorrelationId, linkType + ":" + corrId);
...
}
```

where linkType is either twin, methods, events